### PR TITLE
style(ui): implement dynamic color theme 

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement dynamic color theme

[figma
](https://www.figma.com/file/bcSRcMyKEQc0c7k7QfP9fG/)
image

new added library [color](https://www.npmjs.com/package/color)

## Linear Issue Reference

log-2586

## Testing

test locally

### CODE REVIEW

File: packages/ui/src/components/AppContent/index.module.scss [hunk 1]
```diff
@@ -4,7 +4,17 @@
...
+  --light-hover-variant: rgba(93, 52, 242, 8%);
+  --light-pressed-variant: rgba(93, 52, 242, 12%);
+  --light-hover: #7350f4;
+  --light-pressed: #4718f0;
   --dark-primary-color: #7958ff;
+  --dark-focused-variant: rgba(121, 88, 255, 16%);
+  --dark-hover-variant: rgba(121, 88, 255, 8%);
+  --dark-pressed-variant: rgba(121, 88, 255, 12%);
+  --dark-hover: #957aff;
+  --dark-pressed: #5d36ff;
```
Comment (demonzoo) [inline packages/ui/src/components/AppContent/index.module.scss:17]: Seems these colors are generated in JS now, should we remove them from here?

Comment (simeng-li) [reply packages/ui/src/components/AppContent/index.module.scss:17]: keep a default value

File: packages/ui/src/hooks/use-color-theme.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import color from 'color';
+import { useEffect } from 'react';
+
+// Color hsl lighten/darken takes percentage value only, need to implement absolute value update
+const absoluteLighten = (basColor: color, delta: number) => {
```
Comment (demonzoo) [inline packages/ui/src/hooks/use-color-theme.ts:5]: Just curious what is bas?

**Review by demonzoo — APPROVED** [review]

File: packages/ui/src/hooks/use-color-theme.ts (hunk 2 above)
Comment (IceHe) [reply packages/ui/src/hooks/use-color-theme.ts:5]: baseColor?

Comment (simeng-li) [reply packages/ui/src/hooks/use-color-theme.ts:5]: typo indeed

**Pull request merged**